api/src/main/java/jakarta/data/repository/Query.java

```diff
@@ -34,8 +34,14 @@
  *
  * <p>The required {@link #value} member specifies the JDQL or JPQL query as a string.</p>
  *
- * <p>For {@code select} statements, the return type of the query method must be consistent with the type returned by
- * the query. For {@code update} or {@code delete} statements, it must be {@code void}, {@code int} or {@code long}.</p>
+ * <p>For {@code select} statements, the return type of the query method must be
+ * consistent with the type returned by the query.</p>
+ *
+ * <p>For {@code update} or {@code delete} statements, the return value must be one of:</p>
+ * <ul>
+ * <li>{@code void}</li>
+ * <li>{@code int} or {@code long}, where the value is the number of matching entities</li>
```

Comment:
Please only include that with a NoSQL database that works with eventual persistence; this value might not be as precise as an ACID database. Such as, a relational database.

Reply:
Sure, I added that in https://github.com/jakartaee/data/pull/924/commits/111205c938788370b5b33cdc944cd4e3ee808ae9